### PR TITLE
DEP: integrate: switch simpson to kwarg-only, remove even kwarg

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

#15765
follow-up of https://github.com/scipy/scipy/pull/18209 & https://github.com/scipy/scipy/pull/18843
#### What does this implement/fix?

Scheduled for this release

### CODE REVIEW

File: scipy/integrate/_quadrature.py [hunk 1]
```diff
@@ -605,16 +572,12 @@ def simpson(y, *, x=None, dx=1.0, axis=-1, even=_NoValue):
...
-    romberg : adaptive Romberg quadrature
-    quadrature : adaptive Gaussian quadrature
     fixed_quad : fixed-order Gaussian quadrature
     dblquad : double integrals
     tplquad : triple integrals
     romb : integrators for sampled data
     cumulative_trapezoid : cumulative integration for sampled data
     cumulative_simpson : cumulative integration using Simpson's 1/3 rule
-    ode : ODE integrators
-    odeint : ODE integrators
```
Comment (j-bowhay) [inline scipy/integrate/_quadrature.py:617]: ode solving really has very little to do with simpsons method...

File: scipy/integrate/_quadrature.py [hunk 2]
```diff
@@ -9,8 +9,7 @@
 from scipy.special import roots_legendre
 from scipy.special import gammaln, logsumexp
 from scipy._lib._util import _rng_spawn
-from scipy._lib.deprecation import (_NoValue, _deprecate_positional_args,
-                                    _deprecated)
+from scipy._lib.deprecation import (_deprecated)
```
Comment (lucascolley) [inline scipy/integrate/_quadrature.py:12]: nit: looks a little strange

```suggestion
from scipy._lib.deprecation import _deprecated
```

**Review by lucascolley — APPROVED** [review]
LGTM, checked the previous diffs. Will leave open for a few days

File: scipy/integrate/_quadrature.py (hunk 2 above)
Comment (j-bowhay) [reply scipy/integrate/_quadrature.py]: Blame ruff `--fix` ;)

**Comment by lucascolley** [comment]
thanks Jake

**Pull request merged**